File: src/main/scala/com/vadim/site/model/Student.scala
package com.vadim.site.model

import java.util

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.{DBRef, Document}

import scala.beans.BeanProperty

@Document(collection = "students")
class Student extends Model{

  @Id
  @BeanProperty
  var id: ObjectId = _
  @BeanProperty
  var name: String = _

  @DBRef
  @BeanProperty
  var group: Group = _

  @DBRef
  @BeanProperty
  var books: util.List[Book] = _

  def this(id: ObjectId, name: String, group: Group, books: util.List[Book]){
    this()
    this.id = id
    this.name = name
    this.group = group
    this.books = books
  }

}

File: src/main/scala/com/vadim/site/model/Student.java
package com.vadim.site.model;

import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document(collection = "students")
public class Student implements Model{

    @Id
    private ObjectId id;
    private String name;
    @DBRef
    private Group group;
    @DBRef
    private List<Book> books;

    public Student() {
    }

    public Student(ObjectId id, String name, Group group, List<Book> books) {

        this.id = id;
        this.name = name;
        this.group = group;
        this.books = books;
    }

    public ObjectId getId() {

        return id;
    }

    public void setId(ObjectId id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Group getGroup() {
        return group;
    }

    public void setGroup(Group group) {
        this.group = group;
    }

    public List<Book> getBooks() {
        return books;
    }

    public void setBooks(List<Book> books) {
        this.books = books;
    }
}
